### PR TITLE
test Circle CI file

### PR DESCRIPTION
**Pull request by bam241**
This is an attempt to run CI using in CircleCI

### CODE REVIEW

**Comment by bam241** [comment]
@gonuke @pshriwise @ljacobson64  I think this is ready to be reviewed.

I used a `test_` prefix on the docker image name, so this would need to be changed before we move forward if we decide to...

**Comment by bam241** [comment]
I did there:
- revamp CI to use Circle in place of Travis
- revamp Docker container generation to take advantage of the layering capabilities of the Dockerfile. With this we now have 15 docker containers where we one had 2 before.
- various simplification possible because now each setup has its own docker container.

this should also simplify the addition of new configurations (i.e ubuntu 20.04, other gcc version, other hdf5 version etc...)

File: CI/env.sh
Comment (gonuke) [inline CI/env.sh:18]: Do we actually use this anywhere in CI?

**Comment by bam241** [comment]
I think, some optimisation on the `TRAVIS` variables are possible.

I build a "hack" to populate the same env variable that travis are populating in order to have all the scripts working almost as his. this would probably need to be optimized.... 

but didn't want to spent the time before we actually agree on this.

File: CI/env.sh
Comment (gonuke) [inline CI/env.sh:3]: Is this used?  And why override for GCC below?

**Review by gonuke — CHANGES_REQUESTED** [review]
This looks pretty good to me.  A few questions/comments...

Do you think there would be a way to use Dockerhub to generate all the different variations?

File: CI/env.sh
Comment (bam241) [reply CI/env.sh:3]: gcc compilation crashes if I used all the processors available....

did not dig much more

**Comment by bam241** [comment]
> Do you think there would be a way to use Dockerhub to generate all the different variations?

Probably yes.

It depends on how much time it needs to compile geant4.

but I think we could trigger (in dockerhub) the first build on this repo, and trigger all the other on parents  container modifications.

File: CI/env.sh
Comment (bam241) [reply CI/env.sh:18]: I don't know.
I could try to remove it....

Comment (bam241) [reply CI/env.sh:18]: it is used in our various `build_XXXX.sh` scripts

File: CI/env.sh
Comment (bam241) [reply CI/env.sh:3]: it is used in our various `build_XXXX.sh` scripts

Comment (gonuke) [reply CI/env.sh:3]: Got it.  I guess I was sloppy in looking for where it was used by just searching in the PR on github and didn't think about the other files - oops.

File: CI/env.sh
Comment (gonuke) [reply CI/env.sh:18]: Perhaps it is also used elsewhere in the scripts - I only searched the PR as shown by GH.  However, I don't think we build any fortran in CI or docker???

Comment (bam241) [reply CI/env.sh:18]: I don't think either.

I just put it there, because it looked better, without thinking more about it...

**Comment by bam241** [comment]
our contribution docs and our build status is based on Travis CI, 

While contribution docs could be updated here build status will probably need to wait for the first PR relying on CI to be merged (this one)

File: CI/env.sh
Comment (bam241) [reply CI/env.sh:18]: we do install `gfortran` in docker

Comment (gonuke) [reply CI/env.sh:18]: I think the only place we use FORTRAN is for building Fluka and MCNP.  We don't do either of those in CI (right now).  That said, it might be wise to keep it around in case developers want to use our images as part of their own workflow for building/testing.

File: news/PR-0692.rst
Comment (gonuke) [inline news/PR-0692.rst:1]: This news file probably needs some contents

File: script
Comment (gonuke) [inline script]: Is this meant to be part of the PR?  Doesn't it duplicate what's in `update_docker.sh`?

**Comment by gonuke** [comment]
I think it's OK to wait on the contributing docs until it's easier to create authentic screen shots with Circle CI in use.  I think it's fine to wait on build status as well.

File: script
Comment (bam241) [reply script]: I don't think that was not supposed to be in there.

Comment (gonuke) [reply script]: was your double negative in the above sentence intentional or French?

**Comment by bam241** [comment]
**WARNING**
@gonuke 
I am using a prefix `test_` for all the docker containers using for CircleCI in this PR to avoid collision with the existing containers.
One need to update the name of the containers and fix the circle.yml config file before merging.

Ping me with you feel like this is ready.

File: script
Comment (bam241) [reply script]: Neither I edited my sentence and forgot the second negative...

**Review by gonuke — APPROVED** [review]

**Comment by gonuke** [comment]
I think this is ready, as far as I'm concerned.  Maybe
1. let the CI pass with the current setup
2. rename all the files
3. push those changes
4. let CI pass again
5. I'll merge

**Comment by bam241** [comment]
CI [passed](https://app.circleci.com/pipelines/github/svalinn/DAGMC/98/workflows/2b08e8f8-c733-494f-963a-61e867ce3f1d)

I'll start the update of the docker container.

**Comment by bam241** [comment]
@gonuke I think this is good to go.
CI passed on the renamed Circle Container.

**Comment by gonuke** [comment]
Thanks @bam241

**Pull request merged**
